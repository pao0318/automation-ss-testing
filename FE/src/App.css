.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  padding: 20px;
  color: white;
}

.App-content {
  padding: 20px;
}

nav ul {
  list-style: none;
  padding: 0;
}

nav ul li {
  display: inline-block;
  margin-right: 20px;
}

nav a {
  text-decoration: none;
  color: white;
}

nav button {
  cursor: pointer;
}
