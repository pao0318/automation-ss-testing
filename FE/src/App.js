import React, { useState } from 'react';
import { BrowserRouter as Router, Route, Routes, Navigate, Link } from 'react-router-dom';
import Login from './components/Login/Login.js';
import Homepage from './components/Home/HomePage.js';
import Signup from './components/SignUp/SignUp.js';
import './App.css';

const App = () => {
  const [loggedIn, setLoggedIn] = useState(false);
  const [users, setUsers] = useState([]);

  const handleLogin = () => {
    setLoggedIn(true);
  };

  const handleLogout = () => {
    setLoggedIn(false);
  };

  const handleSignup = (newUsername, newPassword) => {
    setUsers([...users, { username: newUsername, password: newPassword }]);
    setLoggedIn(true);
  };

  return (
    <Router>
      <div className="App">
        <header className="App-header">
          <h1>React Login App</h1>
          <nav>
            <ul>
              <li>
                <Link to="/">Home</Link>
              </li>
              <li>
                {loggedIn ? (
                  <button onClick={handleLogout}>Logout</button>
                ) : (
                  <>
                    <li>
                      <Link to="/login">Login</Link>
                    </li>
                    <li>
                      <Link to="/signup">Signup</Link>
                    </li>
                  </>
                )}
              </li>
            </ul>
          </nav>
        </header>

        <main className="App-content">
          <Routes>
            <Route path="/" element={loggedIn ? <Homepage onLogout={() => setLoggedIn(false)} /> : <Navigate to="/login" />} />
            <Route path="/login" element={<Login onLogin={handleLogin} />} />
            <Route path="/signup" element={<Signup onSignup={handleSignup} />} />
          </Routes>
        </main>
      </div>
    </Router>
  );
};

export default App;

 
