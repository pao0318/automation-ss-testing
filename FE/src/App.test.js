
import React from 'react'
import {render, cleanup} from '@testing-library/react'
import App from './App.js'
import TestElements from './Tests/LoginTest'

 afterEach(cleanup)

 it('should take a snapshot', () => {
    const { asFragment } = render(<App />)

    expect(asFragment(<App />)).toMatchSnapshot()
   })

   it('should test elements from LoginTest', () => {
    render(<TestElements />);
  
    // Use screen.getByText directly
    expect(screen.getByText('Capture and Compare')).toBeInTheDocument();
    // Add more assertions as needed
  });