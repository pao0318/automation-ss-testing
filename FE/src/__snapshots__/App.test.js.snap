// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h1>
        React Login App
      </h1>
      <nav>
        <ul>
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li />
          <li>
            <a
              href="/login"
            >
              Login
            </a>
          </li>
          <li>
            <a
              href="/signup"
            >
              Signup
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main
      class="App-content"
    >
      <div>
        <h1>
          Login
        </h1>
        <form>
          <div>
            <label>
              Username:
              <input
                type="text"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Password:
              <input
                type="password"
                value=""
              />
            </label>
          </div>
          <button
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
